cpp/src/Slice/Parser.h

```diff
@@ -543,26 +543,26 @@ namespace Slice
     {
     public:
         Constructed(const ContainerPtr&, const std::string&);
-        virtual std::string typeId() const;
+        std::string typeId() const override;
     };
 
     // ----------------------------------------------------------------------
     // ClassDecl
     // ----------------------------------------------------------------------
 
-    class ClassDecl : public virtual Constructed
+    class ClassDecl final : public virtual Constructed
     {
     public:
         ClassDecl(const ContainerPtr&, const std::string&);
-        virtual void destroy();
+        void destroy() override;
```

Comment:
Shouldn't all these be final, given ClassDecl is itself final?

Reply:
Yes, you're right. An oversight on my part I guess.